### ISSUE TITLE
Prioritize use of LLMs to improve ordering of pubs, snippet display, and pub based scoring for text-mined results

### ISSUE THREAD

**Issue by mbrush** [NCATSTranslator/Feedback#796]
Filing this ticket in support of the LLM-based 'publication filtering' tool presented by @Rosinaweber at the June 2024 Relay. 

In the example below, four pubs are shown as providing direct support for the text-mined edge "Botulinum Toxin may treat HSP". 

![image](https://github.com/NCATSTranslator/Feedback/assets/5184212/76059096-e0f6-492c-93b9-ff8129f1aaa7)

There are issues with each that demonstrate several known issues. 
- **Pub 1**: text actually reports a lack of evidence for what is asserted in the edge.   
- **Pub 3**: reports that botulism toxin is being studied for its ability to treat HSP - but does not report a conclusion. 
- **Pubs 2 and 4**: the snippets shown does not show the relevant text to determine if/how they do or do not support this edge. Seems to get cut off before we see mention of botulism toxin.  And if you click the link, it takes you to the full pub, where you have to hunt for this text.  Fine, but not ideal.
 
Furthermore, scoring and ranking of this result is based on the assumption that all pubs truly support the edge - which they do not.
 
Examples like these are rampant, and the issues described are well known. I am resurfacing these issues in light of Rosina's proposal at the June 2024 Relay for a 'publication filtering' LLM tool that could help address all of them. Specifically, it could:
 
1. Identify pubs with actual support and show these at the top of the list
2. Count only truly supportive pubs in how a result is scored - give more accurate scores that surface better answers to the top of the list.
3. Help further narrow down or summarize the relevant text to show in the snippets that are displayed in the UI (there may be other solutions to this issue - like making these snippets longer so the relevant text is not cut off)
 
All of these are IMO very useful ways this tool could address important issues. 

I am not informed enough to know how long its development might take, and how hard it would be to incorporate it into how the UI orders/filters/displays pubs and their snippets.  But given the centrality of pub support for users of Translator - I think it might be worth devoting resources to this tool sooner than later - even over the last 6 months of this phase of work.

**Comment by gglusman** [comment]
The question is whether such LLM-based analysis of relevance of each publication is something best done at query time while the user is waiting, and consequently again and again for each publication when the query is repeated... or ahead of time as a way to annotate, classify, or even filter the edges in the underlying KG.

**Comment by webyrd** [comment]
@mbrush thank you for the issue, and thanks @gglusman for the comment.

I brought up this issue on the LLM WG call today.  The consensus of the WG is that use of LLMs for improved NLP, or for improved ranking of publication-based results, is undoubtedly useful.  Given that LitCoin is currently developing a KG that should be useful to Translator, along with the work of the Text Mining Provider team, and the fact that other groups outside of Translator seem to be working on this problem, we think it doesn’t make sense to devote engineering efforts toward this problem during the current funding cycle.  (By the time the next funding cycle begins, such a KG might already have been produced outside of Translator.)

**Comment by Rosinaweber** [comment]
I agree it does not need to be done now but this is a major flaw in Translator's UI.
@webyrd Please let me know if you know of anyone doing this exact task because there are lots of works using LLMs for different tasks. This is an argument-oriented task that is different from link extraction and other similar but sufficiently different tasks. 
@gglusman As other LLM-focused tasks, we need models of correctness to evaluate LLM results and for this reason a memory in support of LLMs selection of prompts needs to be prepared in advance and then executed at runtime. 
Thanks.
